### DISCUSSION TITLE
Can a subscriber layer create new spans?

### DISCUSSION THREAD

**Discussion by miikka** [tokio-rs/tracing#2905] [Q&A] [answered]
A library that I use produces tracing events. I'd like to consume some of these events and replace them with spans to get more informative traces. Is this possible?

I thought I'd create a [Layer](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/layer/index.html), but looks like layers can't create new spans and I'd actually need a Subscriber for that - but Subscribers cannot be easily composed.

**Comment by davidbarsky** [comment, marked as answer]
Not really, and you also run the risk of an infinite span loop: basically, layers aren't able to `map` spans/events.
